vignettes/advanced.Rmd

```diff
@@ -0,0 +1,368 @@
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+    collapse = TRUE,
+    comment = "#>"
+)
+```
+
+
+## Introduction
+
+The purpose of this vignette is to provide an overview of some of the more advanced
+features of the `rbmi` package. 
+
+
+In order to demonstrate these advanced functions we will first create a simulated dataset
+and perform a mock analysis so that all the require objects are available.
+
+```{r}
+library(rbmi)
+library(dplyr)
+set.seed(169)
+dat_full <- simulate_data(n = 120) %>% as_tibble()
+
+## Introduce missingness
+dat <- dat_full
+missing_index_vis2 <- rbinom(nrow(dat), 1, 0.3) == 1 & dat$visit == "visit_2"
+missing_index_vis3 <- rbinom(nrow(dat), 1, 0.4) == 1 & dat$visit == "visit_3"
+dat[missing_index_vis2, "outcome"] <- NA_real_
+dat[missing_index_vis3, "outcome"] <- NA_real_
+
+# create data_ice setting the imputation method to MAR for
+# each patient with at least one missing value
+dat_ice <- dat %>%
+    arrange(id, visit) %>%
+    filter(is.na(outcome)) %>%
+    group_by(id) %>%
+    slice(1) %>%
+    ungroup() %>%
+    select(id, visit) %>%
+    mutate(strategy = "MAR")
+
+
+# Define the names of key variables in our dataset using `set_vars()`
+# Note that covariates argument can contain interactions
+vars <- set_vars(
+    outcome = "outcome",
+    visit = "visit",
+    subjid = "id",
+    group = "group",
+    covariates = c("age", "sex", "group*visit")
+)
+```
+
+
+## Efficiently Changing Imputation Strategies
+
+The `draws()` function is by far the most computationally intensive function in `rbmi`. 
+In some settings, it may be important to explore the impact of a different 
+reference-based imputation strategy than the one chosen for the main analysis. 
+This change in the imputation strategy does not affect the imputation model but it does 
+affect the subsequent imputation step. 
+In order to allow changes in the imputation strategy without having to re-run the 
+`draws()` function, the function `impute()` has an additional argument `update_strategies`.  
+
+Please note though that this functionality comes with a some key limitations:
+From a theoretical point of view, such updates are only sensible if the base imputation model and the data to which it is fitted is also applicable to the revised imputation strategy. For example, as described in the help to the `draws()` function, observed post-ICE data are included in the imputation model for the MAR strategy but excluded for non-MAR strategies. For this reason, the imputation strategy cannot be changed from MAR to a non-MAR strategy via argument `update_strategies` in case there are any observed data after the ICE. A change from a non-MAR to MAR triggers a warning because in this case the imputation model was not fitted to all relevant data and may be inefficient. Similarly, it is not possible to change the first visit which is affected by an ICE via argument `update_strategies`. 
```

Comment:
```suggestion
From a theoretical point of view, such updates are only sensible if the base imputation model and the data to which it is fitted are also applicable to the revised imputation strategy. For example, as described in `help("draws", "rbmi")`, observed post-ICE data are included in the imputation model for the MAR strategy but excluded for non-MAR strategies. For this reason, the imputation strategy cannot be changed from MAR to a non-MAR strategy in the presence of observed data after the ICE. Likewise, a change from a non-MAR to MAR in the presence of observed data after the ICE triggers a warning because the imputation model was not fitted to all relevant data and may be inefficient. Similarly, it is not possible to change the first visit which is affected by an ICE via argument `update_strategies`. 
```